### PR TITLE
docs: Add missing documentations

### LINKED ISSUE #370: Missing github CODEOWNERS file lets approval validation fail
**Description**

When creating a validation for approvals and setting `required: owners` to `true` the validation errors when there is no `.github/CODEOWNERS` file present. Which is a problem as this check is included in the organizational wide `mergeable` settings.

```json
- when: pull_request.*, pull_request_review.*
    name: 'PR has the required number of approvals'
    validate:
      - do: approvals
        min:
          count: 1
        required:
          owners: true
        block:
          changes_requested: true
```

**Actual behavior**

The validation fails with following error message:

```text
HttpError: Resource not accessible by integration
    at /app/node_modules/@octokit/request/dist-node/index.js:66:23
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Job.doExecute (/app/node_modules/bottleneck/light.js:405:18)
```

**Expected behavior**

* The validation does not fail but is ignored instead.

OR

* The validation can be marked as `optional`

### PR DESCRIPTION
**Pull request by shine2lay**
A couple of additions based on feedback and also to make it some minor features more verbose
 
fixes #370

### CODE REVIEW

File: docs/validators/description.rst [hunk 1]
```diff
@@ -24,6 +24,43 @@ Description
...
+
+you can also nest ``and`` and ``or`` options
+
+::
+
+    - do: description
+      and:
+        - or:
+          - must_include:
+              regex: '### Goals'
```
Comment (samj1912) [inline docs/validators/description.rst:55]:
```suggestion
              regex: '### Bug Description'
```

File: docs/validators/description.rst [hunk 2]
```diff
@@ -24,6 +24,43 @@ Description
...
+              message: 'Custom message...'
+          - must_include:
+              regex: '### Changes'
```
Comment (samj1912) [inline docs/validators/description.rst:58]:
```suggestion
              regex: '### Feature Description'
```

File: docs/validators/label.rst [hunk 3]
```diff
@@ -23,6 +23,42 @@ Label
          message: 'Come message...'
       # all of the message sub-option is optional
 
+::
+
+    - do: label
+      and:
+        - must_include:
+            regex: 'DO NOT MERGE'
```
Comment (samj1912) [inline docs/validators/label.rst:31]:
```suggestion
            regex: 'big|medium|small'
```

File: docs/validators/label.rst [hunk 4]
```diff
@@ -23,6 +23,42 @@ Label
...
+            message: 'Custom message...'
+        - must_include:
+            regex: 'type|chore|wont'
+            message: 'Custom message...'
+      or:
+        - must_include:
+            regex: 'type|chore|wont'
```
Comment (samj1912) [inline docs/validators/label.rst:38]:
```suggestion
            regex: 'Ready to merge'
```

File: docs/validators/label.rst [hunk 5]
```diff
@@ -23,6 +23,42 @@ Label
...
+
+you can also nest ``and`` and ``or`` options
+
+::
+
+    - do: label
+      and:
+        - or:
+          - must_include:
+              regex: 'Goals'
```
Comment (samj1912) [inline docs/validators/label.rst:52]:
```suggestion
              regex: 'feat|fix|chore'
```

File: docs/validators/label.rst [hunk 6]
```diff
@@ -23,6 +23,42 @@ Label
...
+              message: 'Custom message...'
+          - must_include:
+              regex: 'Changes'
```
Comment (samj1912) [inline docs/validators/label.rst:55]:
```suggestion
              regex: 'major|minor|patch'
```

File: docs/validators/label.rst [hunk 7]
```diff
@@ -23,6 +23,42 @@ Label
...
+              message: 'Custom message...'
+        - must_include:
+            regex: 'Changes'
```
Comment (samj1912) [inline docs/validators/label.rst:58]:
```suggestion
            regex: 'Ready to merge'
```

File: docs/validators/or.rst [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    - do: and
+      validate:
+        - do: or
+          validate:
+            - do: title
+              begins_with: '[WIP]'
+            - do: label
+              must_include: '[WIP]'
+        - do: label
+          must_include: 'Ready to Merge'
```
Comment (samj1912) [inline docs/validators/or.rst:28]:
```suggestion
          must_include: 'DO NOT MERGE'
```

File: docs/validators/title.rst [hunk 9]
```diff
@@ -23,6 +23,41 @@ Project
          message: 'Come message...'
          # all of the message sub-option is optional
 
+::
+
+    - do: title
+      and:
+        - must_include:
+            regex: '### Goals'
```
Comment (samj1912) [inline docs/validators/title.rst:31]:
```suggestion
            regex: 'ISSUE-\d+'
```

File: docs/validators/title.rst [hunk 10]
```diff
@@ -23,6 +23,41 @@ Project
...
+            message: 'Custom message...'
+        - must_include:
+            regex: '### Changes'
```
Comment (samj1912) [inline docs/validators/title.rst:34]:
```suggestion
            regex: 'type:.+'
```

File: docs/validators/title.rst [hunk 11]
```diff
@@ -23,6 +23,41 @@ Project
...
+            message: 'Custom message...'
+      or:
+        - must_include:
+            regex: '### Goals'
```
Comment (samj1912) [inline docs/validators/title.rst:38]:
```suggestion
            regex: 'feat|chore|fix'
```

File: docs/validators/title.rst [hunk 12]
```diff
@@ -23,6 +23,41 @@ Project
...
+            message: 'Custom message...'
+        - must_include:
+            regex: '### Changes'
```
Comment (samj1912) [inline docs/validators/title.rst:41]:
```suggestion
            regex: 'major|minor|patch'
```

File: docs/validators/title.rst [hunk 13]
```diff
@@ -23,6 +23,41 @@ Project
...
+
+you can also nest ``and`` and ``or`` options
+
+::
+
+    - do: title
+      and:
+        - or:
+          - must_include:
+              regex: '### Goals'
```
Comment (samj1912) [inline docs/validators/title.rst:52]:
```suggestion
              regex: 'feat|fix|chore'
```

File: docs/validators/title.rst [hunk 14]
```diff
@@ -23,6 +23,41 @@ Project
...
+              message: 'Custom message...'
+          - must_include:
+              regex: '### Changes'
```
Comment (samj1912) [inline docs/validators/title.rst:55]:
```suggestion
              regex: 'major|minor|patch'
```

File: docs/validators/title.rst [hunk 15]
```diff
@@ -23,6 +23,41 @@ Project
...
+              message: 'Custom message...'
+        - must_include:
+            regex: '### Changes'
```
Comment (samj1912) [inline docs/validators/title.rst:58]:
```suggestion
            regex: 'ISSUE-\d+'
```

**Review by samj1912 — COMMENTED** [review]
Some minor changes for these to make more sense semantically

File: docs/validators/label.rst (hunk 3 above)
Comment (jusx) [reply docs/validators/label.rst:31]: I like this suggestion. It's a better more realistic example.

File: docs/validators/label.rst (hunk 5 above)
Comment (jusx) [reply docs/validators/label.rst:52]: This is also more realistic example 👍

**Review by jusx — APPROVED** [review]
@shine2lay LGTM. Feel free to merge after considering the suggestions from @samj1912

**Comment by shine2lay** [comment]
@samj1912 thanks for great suggestion, i've applied all of them

**Pull request merged**